onnxruntime/core/providers/dml/DmlExecutionProvider/src/GpuEvent.h

```diff
@@ -3,31 +3,42 @@
 
 #pragma once
 
+#include <wil/wrl.h>
+#include "core/providers/dml/DmlExecutionProvider/src/ErrorHandling.h"
+
 namespace Dml
 {
     // Represents a fence which will be signaled at some point (usually by the GPU).
     struct GpuEvent
     {
         uint64_t fenceValue;
-        ComPtr<ID3D12Fence> fence;
+        Microsoft::WRL::ComPtr<ID3D12Fence> fence;
 
         bool IsSignaled() const
         {
             return fence->GetCompletedValue() >= fenceValue;
         }
 
         // Blocks until IsSignaled returns true.
-        void WaitForSignal() const
+        void WaitForSignal(bool cpuSyncSpinningEnabled) const
         {
             if (IsSignaled())
                 return; // early-out
 
-            wil::unique_handle h(CreateEvent(nullptr, TRUE, FALSE, nullptr));
-            ORT_THROW_LAST_ERROR_IF(!h);
-
-            ORT_THROW_IF_FAILED(fence->SetEventOnCompletion(fenceValue, h.get()));
-
-            WaitForSingleObject(h.get(), INFINITE);
+            if (cpuSyncSpinningEnabled)
+            {
+                while (!IsSignaled())
+                {
+                    // We keep spinning until the fence gets signaled
```

Comment:
No timeout in this `while` loop? Most CPU spinning I've seen has a minimum timeout.

Reply:
I meant to use ORT's `SpinPause` function, forgot to update it.